bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,242 @@
+                main_embed=kata_embed,
+                language_embed=language_embed,
+                tags_embed=tags_embed,
+                other_info_embed=miscellaneous_embed,
+            )
+            await original_message.edit(
+                embed=kata_embed,
+                view=KataView(f'https://codewars.com/kata/{first_kata_id}', dropdown)
+            )
+
+
+# creates the information dropdown
+class InformationDropdown(ui.Select):
+    """A dropdown inheriting from ui.Select that allows finding out other information about the kata."""
+
+    def __init__(
+            self,
+            original_message: Message,
+            language_embed: Embed,
+            tags_embed: Embed,
+            other_info_embed: Embed,
+            main_embed: Embed,
+    ):
+        options = [
+            SelectOption(
+                label="Main Information",
+                description="See the kata's difficulty, description, etc.",
+                emoji='🌎',
+            ),
+            SelectOption(
+                label='Languages',
+                description='See what languages this kata supports!',
+                emoji=Emojis.reddit_post_text,
+            ),
+            SelectOption(
+                label='Tags',
+                description='See what categories this kata falls under!',
+                emoji=Emojis.stackoverflow_tag,
+            ),
+            SelectOption(
+                label='Other Information',
+                description='See how other people performed on this kata and more!',
+                emoji='ℹ',
+            ),
+        ]
+
+        self.original_message = original_message
+        # maps embeds to item chosen
+        self.mapping_of_embeds = {
+            "Main Information": main_embed,
+            "Languages": language_embed,
+            "Tags": tags_embed,
+            "Other Information": other_info_embed,
+        }
+
+        super().__init__(
+            placeholder='See more information regarding this kata',
+            min_values=1,
+            max_values=1,
+            options=options,
+        )
+
+    async def callback(self, interaction: Interaction) -> None:
+        """Callback for when someone clicks on a dropdown."""
+        # option chosen will retrieve the embed for that specific option then the message will be edited
+        result_embed = self.mapping_of_embeds[self.values[0]]
+        await self.original_message.edit(embed=result_embed)
+
+
+# creating the discord.ui.View subclass for the UI in the challenges command.
+class KataView(ui.View):
+    """The view for the InformationDropdown and the link to the Kata."""
+
+    def __init__(self, link: str, dropdown: InformationDropdown):
+        super().__init__()
+
+        # adding the dropdown
+        self.add_item(dropdown)
+        # manually creating the button as it is a link button and has no decorator
+        self.add_item(ui.Button(label="View the Kata", url=link))
```

Comment:
We don't use the functionality of subclassing particularly much, we can just create an instance directly.
```py
view = ui.View()
view.add_item(dropdown)
view.add_item(ui.Button(label="View the kata", url=link))
```
Just place this in the command body itself

Reply:
Sounds good

Reply:
@Bluenix2 Alright, I made the command much smaller!
Let me know what you think when you're free!